core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -353,13 +355,26 @@ public void close(Throwable cause) {
         }
 
         private void close(@Nullable Throwable cause, boolean cancel) {
+            if (closed) {
+                return;
+            }
+            closed = true;
             state = State.DONE;
             cancelTimeoutOrLog(cause, cancel);
             if (ctx != null) {
-                if (cause == null) {
-                    ctx.request().abort(ResponseCompleteException.get());
-                } else {
-                    ctx.request().abort(cause);
+                final HttpRequest request = ctx.request();
+                assert request != null;
+                if (cause != null) {
+                    request.abort(cause);
+                    return;
+                }
+                if (ctx.requestAutoAbortDelayMillis() == 0 || !request.isOpen()) {
+                    request.abort(ResponseCompleteException.get());
+                    return;
+                }
+                if (ctx.requestAutoAbortDelayMillis() > 0) {
```

Comment:
`Long.MAX_VALUE` can be considered the equivalent of disabling the auto abortion.
```suggestion
                if (ctx.requestAutoAbortDelayMillis() > 0 && ctx.requestAutoAbortDelayMillis() < Long.MAX_VALUE) {
```

Reply:
Nice catch!